### ISSUE TITLE
How to build only SILK part of Opus codec for low-level use (such as encode_pulses, decode_pulses)

### ISSUE THREAD

**Issue by Ori248** [xiph/opus#276]
Hello. I'm trying to implement a certain scheme that requires a low level use of the Opus codec, specifically the SILK part of it. Even more specifically, it requires a use of encode_pulses.c and decode_pulses.c. In addition, I need this part to be compiled into a shared object file (.so file), so that I can call these functions from my Python script. When I follow the build instructions given in the Opus GitHub, I get the shared file "libopus.so", and when I view the list of functions in it, it doesn't contain the desired low-level functionaliy I need. My question is, is there an option to build a specific part of the Opus codec, so that low-level functionality can be used directly from a Python script using ctypes? Thanks in advance

**Comment by xnorpx** [comment]
No, you would need to create your own dll and export the types you want on top the lib.
